lightning/src/ln/onion_utils.rs

```diff
@@ -1650,6 +1650,34 @@ impl From<u16> for LocalHTLCFailureReason {
 	}
 }
 
+impl_writeable_tlv_based_enum!(LocalHTLCFailureReason,
+	(1, TemporaryNodeFailure) => {},
```

Comment:
Commit message clear now. Is this type of empty value tlv structure the way to serialize the enum though? I think I was expecting some internal code for serialization specifically. Might be worth a code comment.

Reply:
> Is this type of empty value tlv structure the way to serialize the enum though?

afaik yes, it's serializing a struct enum with no fields. If we wanted to add more context to each of these variants, we'd be able to add them as TLVs here. 

There aren't a lot of enums in the codebase that don't have extra data, but `BlindedFailure` is done like this too.